allennlp/data/data_loaders/multitask_epoch_sampler.py

```diff
@@ -0,0 +1,67 @@
+from typing import Any, Dict, Mapping
+
+from allennlp.common.registrable import Registrable
+from allennlp.data.data_loaders.data_loader import DataLoader
+
+
+class MultiTaskEpochSampler(Registrable):
+    """
+    A class that determines with what proportion each dataset should be sampled for a given epoch.
+    This is used by the `MultiTaskDataLoader`.  The main output of this class is the task proportion
+    dictionary returned by `get_task_proportions`, which specifies what percentage of the instances
+    for the current epoch should come from each dataset.  To control this behavior as training
+    progresses, there is an `update_from_epoch_metrics` method, which should be called from a
+    `Callback` during training.
+    """
+
+    def get_task_proportions(self, data_loaders: Mapping[str, DataLoader]) -> Dict[str, float]:
+        """
+        Given a dictionary of `DataLoaders` for each dataset, returns what percentage of the
+        instances for the current epoch of training should come from each dataset.  The input
+        dictionary could be used to determine how many datasets there are (e.g., for uniform
+        sampling) or how big each dataset is (e.g., for sampling based on size), or it could be
+        ignored entirely.
+        """
+        raise NotImplementedError
```

Comment:
This would be a safer API if the caller had to normalize the proportions so they sum up to 1. Right now, if this sums up to 2, your epoch will have 2x the expected number of instances.

Reply:
Done.